### PR TITLE
[CELEBORN-1425][HELM] Add helm chart unit tests to ensure manifests are rendered as expected

### PR DESCRIPTION
**Pull request by ChenYi015**
### What changes were proposed in this pull request?

Add helm chart unit tests.

### Why are the changes needed?

Unit tests can make resource manifests are rendered as expected with various configurations.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Detailed information about how to run helm chart unit tests can be found here [helm-unittest/helm-unittest](https://github.com/helm-unittest/helm-unittest). First, you need to install helm unit test plugin:

```shell
helm plugin install https://github.com/helm-unittest/helm-unittest.git
```

Then, run helm chart unitt tests as follows:

```shell
$ helm unittest charts/celeborn  --file "tests/**/*_test.yaml" --strict --debug
load_plugins.go:110: [info] file (/Users/chenyi/Library/helm/plugins/helm-acr/completion.yaml) not provided by plugin. No plugin auto-completion possible

### Chart [ celeborn ] charts/celeborn

 PASS  Test Celeborn configmap  charts/celeborn/tests/configmap_test.yaml
 PASS  Test Celeborn master pod monitor charts/celeborn/tests/master/podmonitor_test.yaml
 PASS  Test Celeborn master priority class      charts/celeborn/tests/master/priorityclass_test.yaml
 PASS  Test Celeborn master service     charts/celeborn/tests/master/service_test.yaml
 PASS  Test Celeborn master statefulset charts/celeborn/tests/master/statefulset_test.yaml
 PASS  Test Celeborn worker pod monitor charts/celeborn/tests/worker/podmonitor_test.yaml
 PASS  Test Celeborn worker priority class      charts/celeborn/tests/worker/priorityclass_test.yaml
 PASS  Test Celeborn worker service     charts/celeborn/tests/worker/service_test.yaml
 PASS  Test Celeborn worker statefulset charts/celeborn/tests/worker/statefulset_test.yaml

Charts:      1 passed, 1 total
Test Suites: 9 passed, 9 total
Tests:       48 passed, 48 total
Snapshot:    0 passed, 0 total
Time:        183.011375ms

```

### CODE REVIEW

**Comment by ChenYi015** [comment]
@pan3793 PTAL, thanks!

**Review by pan3793 — APPROVED** [review]
That's cool.

**Comment by pan3793** [comment]
could you please also write dev docs to guide contributors on how to run unit test? and set up CI for that? You can do it in dedicated PR or do it in place

**Comment by ChenYi015** [comment]
> could you please also write dev docs to guide contributors on how to run unit test? and set up CI for that? You can do it in dedicated PR or do it in place

Ok, I will update docs in place and setup CI in a dedicated PR.

**Comment by ChenYi015** [comment]
@pan3793 I had added  deverloper docs for running helm chart unit tests.

File: docs/developers/helm-charts.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+---
+
+# Helm Charts
+
+## Run Helm Unit Tests
+
+To run unit tests against Helm charts, first you need to install the helm unittest plugin as follows:
+
+```shell
+helm plugin install https://github.com/helm-unittest/helm-unittest.git
```
Comment (pan3793) [inline docs/developers/helm-charts.md:26]: Does that mean always using the latest version? it's fragile

File: docs/developers/helm-charts.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
```
Comment (pan3793) [inline docs/developers/helm-charts.md:1]: when adding new pages, `mkdocs.yml` should be updated too.
https://celeborn.apache.org/community/contributor_guide/docs_and_website/

File: docs/developers/helm-charts.md (hunk 1 above)
Comment (ChenYi015) [reply docs/developers/helm-charts.md]: > Does that mean always using the latest version? it's fragile

@pan3793 Yes, have updated to use fixed version `0.5.1`.

File: docs/developers/helm-charts.md (hunk 2 above)
Comment (ChenYi015) [reply docs/developers/helm-charts.md:1]: @pan3793 Updated.

**Comment by ChenYi015** [comment]
@pan3793 I have updated the docs, PTAL, thanks.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Also cc @RexXiong

**Comment by pan3793** [comment]
Thanks, merged to main for 0.5

**Pull request closed without merging**